actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,294 @@
+        distributed_executor_address: Optional[str] = None,
+        overwrite: bool = False,
+        metadata: Optional[str] = None,
+        feature: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Run a pure function.
+
+        Protected Parameters
+        --------------------
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+        clean: bool
+            Should the local staging directory be cleaned prior to this run.
+            Default: False (Do not clean)
+        debug: bool
+            A debug flag for the developer to use to manipulate how much data runs,
+            how it is processed, etc.
+            Default: False (Do not debug)
+
+        Parameters
+        ----------
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame]
+            The primary cell dataset to use for generating 
+            diagnistic sheet for a group of cells.
+
+            **Required dataset columns:** *["CellId"]*
+
+        metadata: str
+            The metadata to group cells and generate a diagnostic sheet. 
+            For example, "FOVId"
+
+        feature: str
+            The name of the single cell feature to display. For example, "imsize_orig"
+
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+            Default: None
+
+        Returns
+        -------
+        manifest_save_path: Path
+            Path to the produced manifest with the DiagnosticSheetPath column added.
+        """
+        if isinstance(dataset, (str, Path)):
+            dataset = Path(dataset).expanduser().resolve(strict=True)
+
+            # Read dataset
+            dataset = pd.read_csv(dataset)
+
+        # Check dataset and manifest have required fields
+        dataset_utils.check_required_fields(
+            dataset=dataset, required_fields=REQUIRED_DATASET_FIELDS,
+        )
+
+        # Create save directories
+        diagnostic_sheet_dir = self.step_local_staging_dir
+        diagnostic_sheet_dir.mkdir(exist_ok=True)
+
+        # Process each row
+        with DistributedHandler(distributed_executor_address) as handler:
+            # Start processing
+            diagnostic_sheet_futures = handler.client.map(
+                self._collect_group,
+                # Convert dataframe iterrows into two lists of items to iterate over
+                # One list will be row index
+                # One list will be the pandas series of every row
+                *zip(*list(dataset.iterrows())),
+                [diagnostic_sheet_dir for i in range(len(dataset))],
+                [overwrite for i in range(len(dataset))],
+                [metadata for i in range(len(dataset))],
+            )
+
+            # Block until all complete
+            diagnostic_sheet_result = handler.gather(diagnostic_sheet_futures)
```

Comment:
When you merge `master` you will see that across the board we now use `handler.batched_map`.

It doesn't return futures it returns the results themselves but you should also pass the `batch_size` parameter in. See above comment.